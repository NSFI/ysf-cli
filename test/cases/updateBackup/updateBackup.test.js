const path = require("path");
const fsExtra = require("fs-extra");
const shell = require("shelljs");
const glob = require("glob");
describe("UpdateMethod", () => {
  beforeAll(() => {
    // 重建测试环境

    process.chdir(__dirname); // jest.mock path resolve
    shell.rm("-rf", "./testOutput");
    shell.cp("-R", "./reset", "./testOutput");
  });
  it("should backup files", async () => {
    const variables = require("../../../src/update/variables");
    variables.PROJECT_DIR = path.resolve("./testOutput");
    variables.CACHE_DIR = path.resolve("./");

    //  prepare
    const updateMethod = require("../../../src/update/updateMethods");

    const updateConfig = await fsExtra.readJSON(
      path.resolve(__dirname, "./u/update.json")
    );

    //  execute
    await updateMethod.backup(updateConfig);

    //  verify
    let expectedFiles = await glob.sync("testOutput/**", {
      cwd: __dirname,
    });

    expect(expectedFiles).toHaveLength(11);
    expect(expectedFiles).toMatchSnapshot();
  });
  afterAll(() => {
    jest.resetAllMocks();
  });
});
