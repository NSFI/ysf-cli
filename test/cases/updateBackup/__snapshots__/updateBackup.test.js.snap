// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateMethod should backup files 1`] = `
Array [
  "testOutput",
  "testOutput/fileA",
  "testOutput/fileA.bak",
  "testOutput/fileB",
  "testOutput/fileB.bak",
  "testOutput/fileC",
  "testOutput/fileC.bak",
  "testOutput/folderToBackup",
  "testOutput/folderToBackup.bak",
  "testOutput/folderToBackup.bak/delteFiile",
  "testOutput/folderToBackup/delteFiile",
]
`;
